tasks/sqlcmd_input_file.yml

```
@@ -42,11 +42,15 @@
         or
         (__mssql_sqlcmd_input is failed)
       block:
-        - name: >-
-            Print the output of the sqlcmd command
-            if not empty for {{ item }}
+        - name: Print the output of the sqlcmd command for {{ __mssql_debug }}
+          when: "{{ __mssql_debug }} | length > 0"  # noqa no-jinja-when
```

```suggestion
          when: __mssql_debug | length > 0
```
unless I'm missing something?

---

does `__mssql_debug` hold the name of another variable you want to expand?  If so, I think using the `vars` lookup https://docs.ansible.com/ansible/2.9/plugins/lookup/vars.html is a better way to do this.